.github/workflows/lint.yml

```diff
@@ -2,17 +2,55 @@ on:
   workflow_call:
     inputs:
       type:
-        description: "the expected lint entrypoint: a language name or either 'make' or 'just'"
-        default: "make"
+        description: "the expected lint entrypoint: make, just, or github"
+        default: "github"
+        required: false
+        type: string
+      language:
+        description: "the language to configure"
+        required: true
+        type: string
+      continue-on-error:
+        description: "don't mark the job as failed if a lint fails"
+        default: false
+        required: false
+        type: boolean
+      target:
+        description: "the make or just target to run for linting"
+        default: "lint"
         required: false
         type: string
       python-version:
         description: "the version of Python to configure, if any (can be SemVer-formatted)"
         default: ""
         required: false
         type: string
+      python-version-file:
+        description: "get the Python version from the specified file"
+        default: ".python-version"
+        required: false
+        type: string
+      python-cache-dependency-path:
+        description: "the file(s) where dependencies are defined: pyproject.toml, requirements.txt, poetry.lock"
+        default: |
+          **/pyproject.toml
+          **/requirements*.txt
+          **/Pipfile.lock
+          **/poetry.lock
```

Comment:
Small thing, but maybe it makes sense to be normative here and exclude `Pipfile.lock` and `poetry.lock`? These indicate a project that we haven't ported over to `pyproject.toml` yet, so including them in the cache check suggests support when we probably want to (gently!) discourage them 🙂